src/test/java/io/specto/hoverfly/ruletest/HoverflyDslWithStateTest.java

```diff
@@ -0,0 +1,82 @@
+package io.specto.hoverfly.ruletest;
+
+import io.specto.hoverfly.junit.rule.HoverflyRule;
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.client.HttpServerErrorException;
+import org.springframework.web.client.RestTemplate;
+
+import static io.specto.hoverfly.junit.core.SimulationSource.dsl;
+import static io.specto.hoverfly.junit.dsl.HoverflyDsl.service;
+import static io.specto.hoverfly.junit.dsl.ResponseCreators.created;
+import static io.specto.hoverfly.junit.dsl.ResponseCreators.success;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatExceptionOfType;
+
+public class HoverflyDslWithStateTest {
+
+    @ClassRule
+    public static HoverflyRule hoverflyRule = HoverflyRule.inSimulationMode(dsl(
+        service("www.service-with-state.com")
+            .get("/api/bookings/create")
+            .willReturn(success().transitionsState("hasBeenAlreadyCreated", "true"))
+
+            .get("/api/bookings/create")
+            .requiresState("hasBeenAlreadyCreated", "true")
+            .willReturn(created())
+
+            .get("/api/bookings/remove")
+            .requiresState("hasBeenAlreadyCreated", "true")
+            .willReturn(success().removesState("hasBeenAlreadyCreated"))
+
+            .get("/remove-without-required-state")
+            .willReturn(success().removesState("hasBeenAlreadyCreated"))
+    )).printSimulationData();
+
+    private final RestTemplate restTemplate = new RestTemplate();
+
+    @Before
+    public void removeState(){
```

Comment:
Is this for resetting all the states before each test? It'd be better to add a `resetStates` method to `HoverflyRule` by calling this API: http://hoverfly.readthedocs.io/en/latest/pages/reference/api/api.html#delete-api-v2-state